### PR TITLE
Add Fireworks AI as new model provider

### LINKED ISSUE #8424: Add Fireworks llm model provider to support
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing? Tell me about your story.

[Fireworks AI](https://fireworks.ai/) delivers a fast, affordable, and customizable platform for developers to run and fine-tune generative artificial intelligence (AI) models at scale.

### 2. Additional context or comments

_No response_

### 3. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

### PR DESCRIPTION
**Pull request by yaoice**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Fixes #8424

fireworks model references: https://fireworks.ai/models

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

image
image
image
image
image
image

### CODE REVIEW

File: api/core/model_runtime/model_providers/fireworks/_assets/icon_l_en.svg
Comment (crazywoola) [inline api/core/model_runtime/model_providers/fireworks/_assets/icon_l_en.svg:1]: It seems that the logo is breaking the UI

**Review by crazywoola — CHANGES_REQUESTED** [review]
See comments

File: api/core/model_runtime/model_providers/fireworks/_assets/icon_l_en.svg
Comment (yaoice) [reply api/core/model_runtime/model_providers/fireworks/_assets/icon_l_en.svg:1]: I hava update the icon_l_en.svg to fit the UI as following：
image

**Comment by yaoice** [comment]
> See comments

@crazywoola I have done, pls review it again.

**Review by crazywoola — APPROVED** [review]

**Pull request merged**